### PR TITLE
Added Docker build action

### LINKED ISSUE #8105: Add Docker build Action
It should be relatively easy to build Docker images using GitHub Actions, taken from https://docs.docker.com/build/ci/github-actions/share-image-jobs/ :

```yml
name: ci

on:
  push:

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3
      
      - name: Build and export
        uses: docker/build-push-action@v6
        with:
          tags: myimage:latest
          outputs: type=docker,dest=/tmp/myimage.tar
      
      - name: Upload artifact
        uses: actions/upload-artifact@v4
        with:
          name: myimage
          path: /tmp/myimage.tar
```

If we automatically publish images, we also do not have to worry about maintaining (as much) documentation.

### PR DESCRIPTION
**Pull request by qstokkink**
Fixes #8105

This PR:

 - Adds a Docker image build action.

The latest build is available here: https://github.com/qstokkink/tribler/actions/runs/10506075807

### CODE REVIEW

**Comment by qstokkink** [comment]
validate

**Comment by xoriole** [comment]
The docker image is requiring the DISPLAY, and is failing with the following stacktrace:
```python
Traceback (most recent call last):
  File "/app/tribler/src/run_tribler.py", line 14, in <module>
    import pystray
  File "/usr/local/lib/python3.10/site-packages/pystray/__init__.py", line 64, in <module>
    Icon = backend().Icon
  File "/usr/local/lib/python3.10/site-packages/pystray/__init__.py", line 56, in backend
    return candidate()
  File "/usr/local/lib/python3.10/site-packages/pystray/__init__.py", line 36, in xorg
    from . import _xorg as backend; return backend
  File "/usr/local/lib/python3.10/site-packages/pystray/_xorg.py", line 36, in <module>
    display = Xlib.display.Display()
  File "/usr/local/lib/python3.10/site-packages/Xlib/display.py", line 89, in __init__
    self.display = _BaseDisplay(display)
  File "/usr/local/lib/python3.10/site-packages/Xlib/display.py", line 71, in __init__
    protocol_display.Display.__init__(self, *args, **keys)
  File "/usr/local/lib/python3.10/site-packages/Xlib/protocol/display.py", line 84, in __init__
    name, protocol, host, displayno, screenno = connect.get_display(display)
  File "/usr/local/lib/python3.10/site-packages/Xlib/support/connect.py", line 73, in get_display
    return mod.get_display(display)
  File "/usr/local/lib/python3.10/site-packages/Xlib/support/unix_connect.py", line 59, in get_display
    raise error.DisplayNameError(display)
Xlib.error.DisplayNameError: Bad display name ""

```

**Comment by qstokkink** [comment]
That should be resolved by https://stackoverflow.com/a/44434831

**Comment by qstokkink** [comment]
Another issue: this build does not package the `npm` output (yet).

File: .github/workflows/build_docker.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+name: Publish Docker Image
+
+on:
```
Comment (xoriole) [inline .github/workflows/build_docker.yml:3]: @qstokkink 
Which triggers are expected for the docker build?

Comment (qstokkink) [reply .github/workflows/build_docker.yml:3]: Just the `workflow_dispatch`. Eventually, I want this to be part of the `build.yml` to have a stable Docker image per release.

**Comment by qstokkink** [comment]
validate

**Review by xoriole — APPROVED** [review]

**Comment by qstokkink** [comment]
validate

**Comment by qstokkink** [comment]
validate

**Review by egbertbouman — APPROVED** [review]

**Pull request merged**